pkg/image/resolve_test.go

```diff
@@ -191,6 +192,84 @@ func TestResolveWithCluster(t *testing.T) {
 	}
 }
 
+func TestGetImages(t *testing.T) {
+	// Setup a httpserver
+	server := setupTestServer()
+	defer server.Close()
+
+	// Mock the parseRepoName function to return the expected repoName
+	originalParseRepoName := parseRepoName
+	parseRepoName = func(url string) (string, error) {
+		return "rancher/ui-plugin-catalog", nil
+	}
+	defer func() {
+		parseRepoName = originalParseRepoName
+	}()
+
+	tests := []struct {
+		name         string
+		exportConfig ExportConfig
+		expected     []string
+		notExpected  []string
+	}{
+		{
+			name: "exportConfig is completely empty",
+			exportConfig: ExportConfig{
+				SystemChartsPath: "",
+				ChartsPath:       "",
+				OsType:           Linux,
+				RancherVersion:   "",
+				GithubEndpoints:  []GithubEndpoint{},
+			},
+			expected:    []string{},
+			notExpected: []string{"rancher/ui-plugin-catalog"},
+		},
+		{
+			name: "only extensions is set in exportConfig",
+			exportConfig: ExportConfig{
+				SystemChartsPath: "",
+				ChartsPath:       "",
+				OsType:           Linux,
+				RancherVersion:   "",
+				GithubEndpoints: []GithubEndpoint{
+					{URL: server.URL},
+				},
+			},
+			expected:    []string{"rancher/ui-plugin-catalog"},
+			notExpected: []string{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			imagesList, _, err := GetImages(tt.exportConfig, make(map[string][]string), []string{}, map[string]rketypes.RKESystemImages{})
+			assertlib.NoError(t, err)
+
+			for expected := range tt.expected {
+				found := false
+				for image := range imagesList {
+					if strings.Contains(imagesList[image], tt.expected[expected]) {
+						found = true
+					}
+				}
+				assertlib.True(t, found)
+			}
```

Comment:
Minor nit, but I think we can update iteration to use values rather than indexes for a bit cleaner code. We could also break as soon as the match is found. Same can be done for the iteration below.

Maybe there is something I'm missing why we have to use indexes here.

```suggestion
			for _, expected := range tt.expected {
				found := false
				for _, image := range imagesList {
					if strings.Contains(image, expected) {
						found = true
						break
					}
				}
				assertlib.True(t, found)
			}
```